cmake/IgnPython.cmake

```diff
@@ -0,0 +1,25 @@
+# Copyright 2020 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Copied from ament/ament_cmake: ament_cmake/ament_cmake_core/cmake/core/python.cmake
+
+set(PYTHON_VERSION "" CACHE STRING
+  "Specify specific Python version to use ('major.minor' or 'major')")
+
+# if not specified otherwise use Python 3
+if(NOT PYTHON_VERSION)
+  set(PYTHON_VERSION "3")
+endif()
+
+find_package(PythonInterp ${PYTHON_VERSION} REQUIRED)
```

Comment:
does this mean `python` is now a required dependency of `ignition-cmake2`? the homebrew bottle build of 2.6.0 in https://github.com/osrf/homebrew-simulation/pull/1222 is failing

[![Build Status](https://build.osrfoundation.org/buildStatus/icon?job=generic-release-homebrew_triggered_bottle_builder%2Flabel%3Dosx_mojave&build=124)](https://build.osrfoundation.org/job/generic-release-homebrew_triggered_bottle_builder/label=osx_mojave/124/) https://build.osrfoundation.org/job/generic-release-homebrew_triggered_bottle_builder/label=osx_mojave/124/

Reply:
I think we might want this to be `QUIET` instead of `REQUIRED`

Reply:
> does this mean python is now a required dependency of ignition-cmake2?

I think that this is what this line implies, although I'm not sure if we meant to create this dependency or not.

> I think we might want this to be QUIET instead of REQUIRED

If we do need python though, what would the implications be of replacing `REQUIRED` with `QUIET`? I'm wondering if marking this `QUIET` can cause issues elsewhere if python is not found. @mjcarroll, what are your thoughts?

Reply:
If it's an optional dependency that's only needed for optional modules (e.g. testing?), then whatever modules depend on it should check `${PythonInterp_FOUND}` to conditionally build only if this package was found. If the package was found, then the module should skip being built.

Reply:
we can continue the discussion in https://github.com/ignitionrobotics/ign-cmake/pull/132, which proposes to revert python3 to an optional dependency